state/syncer/userAccountsSyncer.go

```diff
@@ -256,6 +268,7 @@ func (u *userAccountsSyncer) syncAccountDataTries(
 			if newErr != nil {
 				errMutex.Lock()
 				errFound = newErr
```

Comment:
I think `errMutex`, `errFound` can be removed, and instead use the `ErrChan` from `leavesChannels`.